packages/react-instantsearch/examples/autocomplete/src/App-Hits-And-Facets.js

```diff
@@ -0,0 +1,143 @@
+    this.state = {searchState: {}, hits: {}, query: ''};
+  }
+
+  onSearchStateChange(searchState) {
+    this.setState({searchState: {...this.state.searchState, ...searchState}});
+  }
+
+  onProps(props) {
+    props.facets[props.indexName] = props.hits;
+    this.setState({hits: {...this.state.hits, ...props.facets}, query: props.query});
+  }
+
+  formatHitsForAutoSuggest() {
+    const hits = [];
+    forOwn(this.state.hits, (value, key) => {
+      hits.push({title: key, hits: value});
+    });
+    return hits;
+  }
+
+  render() {
+    return (
+      <div>
+        <FirstResults onProps={this.onProps.bind(this)}
+                      searchState={this.state.searchState}
+                      onSearchStateChange={this.onSearchStateChange.bind(this)}/>
+        <InstantSearch
+          appId="latency"
+          apiKey="6be0576ff61c053d5f9a3225e2a90f76"
+          indexName="bestbuy"
+        >
+          <Autosuggest
+            suggestions={this.formatHitsForAutoSuggest()}
+            multiSection={true}
+            onSuggestionsFetchRequested={({value}) => this.onSearchStateChange({query: value})}
+            onSuggestionsClearRequested={() => this.onSearchStateChange({query: ''})}
+            getSuggestionValue={hit => hit.name}
+            renderSuggestion={hit =>
+              <div>
+                <div>{hit.name}</div>
+              </div>
+            }
+            inputProps={{
+              placeholder: 'Type a product',
+              value: this.state.query,
+              onChange: () => {
+              },
+            }}
+            renderSectionTitle={section => section.title}
+            getSectionSuggestions={section => section.hits}
+          />
+        </InstantSearch>
+      </div>
+    );
+  }
+}
+
+class FirstResults extends React.Component {
+  shouldComponentUpdate(nextProps) {
+    return !isEqual(this.props.searchState, nextProps.searchState);
+  }
+
+  render() {
+    return <InstantSearch
+      appId="latency"
+      apiKey="6be0576ff61c053d5f9a3225e2a90f76"
+      indexName="bestbuy"
+      searchState={this.props.searchState}
+      onSearchStateChange={this.props.onSearchStateChange}
+      searchParameters={{hitsPerPage: 3}}
+    >
+      <div>
+        <VirtualSearchBox/>
+        <VirtualRefinementList attributeName="manufacturer"/>
+        <VirtualRefinementList attributeName="category"/>
+        <VirtualAutoSuggest indexName="product" onProps={this.props.onProps} attributes={['manufacturer', 'category']}/>
+      </div>
+    </InstantSearch>;
+  }
+}
+
+FirstResults.propTypes = {
+  searchState: React.PropTypes.object.isRequired,
+  onSearchStateChange: React.PropTypes.func.isRequired,
+  onProps: React.PropTypes.func.isRequired,
+};
+
+const VirtualSearchBox = connectSearchBox(() => null);
+const VirtualRefinementList = connectRefinementList(() => null);
+
+const connectAutoComplete = createConnector({
+  displayName: 'AutoComplete',
+
+  getProvidedProps(props, state, search) {
```

Comment:
We use a custom connector here to be able to gather every props we need before sending them to the higher component. We need to gather all those data because the external autocomplete component (here react-autosuggest) needs to be initialize with the entire state at once.